BGAnimations/ScreenEvaluation common/PerPlayer/Lower/Graphs.lua

```diff
@@ -54,6 +54,20 @@ af[#af+1] = Def.GraphDisplay{
 		local ColorIndex = ((SL.Global.ActiveColorIndex + (player==PLAYER_1 and -1 or 1)) % #SL.Colors) + 1
 		self:Load("GraphDisplay" .. ColorIndex )
 
+		local firstSecond = math.min(TimingData:GetElapsedTimeFromBeat(0), 0)
+		local chartStartSecond = GAMESTATE:GetCurrentSong():GetFirstSecond()
+		local lastSecond = GAMESTATE:GetCurrentSong():GetLastSecond()
+		local duration = lastSecond - firstSecond
+
+		-- GraphDisplay starts at chartStartSecond, but the NPS graph
+		-- and the scatter plot start at firstSecond, so we have to
+		-- move the lifebar to the correct offset to align it with the
+		-- NPS graph.
+		local offsetFactor = (chartStartSecond - firstSecond) / duration
```

Comment:
For confirmation, does this work with ratemods?

Reply:
Yeah, it does!